ln-gateway/src/messaging.rs

```diff
@@ -0,0 +1,44 @@
+use std::sync::Arc;
+
+use anyhow::Error;
+use async_trait::async_trait;
+use tokio::sync::{mpsc, oneshot, Mutex};
+
+use crate::{GatewayRequest, GatewayRequestTrait, Result};
+
+#[async_trait]
+pub trait GatewayMessageReceiver: Send + Sync + 'static {
+    async fn receive(&self) -> ();
+}
+
+#[derive(Debug, Clone)]
+pub struct GatewayMessageChannel {
+    sender: Arc<Mutex<mpsc::Sender<GatewayRequest>>>,
```

Comment:
`Arc<Mutex<…>>` is not useful here. `mpsc::Sender` is already a synchronization primitive and implements `Clone`. Clones of a sender can be sent between threads etc.